R/test-compiled-code.r

```diff
@@ -0,0 +1,121 @@
+# Geenrates a (semi-random) directory wherein compilation
+# can take place, within tempdir().
+cpp_compilation_tempdir <- function(package) {
+
+  # Generate a path wherein we perform the executable compilation
+  random_string <- paste(
```

Comment:
I wasn't aware of the `pattern` argument to `tempfile()` -- I'll use that instead.